redfish-core/lib/log_services.hpp

```diff
@@ -3411,6 +3436,610 @@ inline void getSystemHardwareIsolationLogService(
+                if (ec || objects.empty())
+                {
+                    BMCWEB_LOG_ERROR << "Failed to get parent DBus path for "
+                                        "identified parent Redfish URI: "
+                                     << parentRedfishUri
+                                     << " of the given DBus object path: "
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                std::string uriIdObj;
+                for (auto& object : objects)
+                {
+                    if (dbusObjPath.str.find(object) == std::string::npos)
+                    {
+                        continue;
+                    }
+                    uriIdObj = object;
+                    break;
+                }
+
+                if (uriIdObj.empty())
+                {
+                    BMCWEB_LOG_ERROR << "Failed to get parent DBus path for "
+                                        "identified parent Redfish URI: "
+                                     << parentRedfishUri
+                                     << " of the given DBus object path: "
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                redfishUri.replace(
+                    uriIdPos, uriIdPattern.length(),
+                    getIsolatedHwItemId(
+                        sdbusplus::message::object_path(uriIdObj)));
+
+                if (redfishUri.find(uriIdPattern) == std::string::npos)
+                {
+                    nlohmann::json& entryJson =
+                        (entryJsonIdx > 0
+                             ? asyncResp->res
+                                   .jsonValue["Members"][entryJsonIdx - 1]
+                             : asyncResp->res.jsonValue);
+
+                    entryJson["Links"]["OriginOfCondition"] = {
+                        {"@odata.id", redfishUri}};
+                }
+                else
+                {
+                    fillRedfishUriWithIds(asyncResp, entryJsonIdx, dbusObjPath,
+                                          redfishUri);
+                }
+            },
+            objMapperSerName, objMapperObjPath, objMapperIface,
+            "GetSubTreePaths", "/xyz/openbmc_project/inventory", 0,
+            std::get<0>(*parentRedfishUriIt));
+    }
+}
+
+/**
+ * @brief API used to fill the respective redfish uri (which will get by using
+ *        the given dbus object path) into "OriginOfCondition" property of
+ *        LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] redfishUri - The redfish uri to fill with its "Id".
```

Comment:
redfishUri - The redfish uri to fill with its "Id". ===> this seems to be an iterator